File: src/main/java/CreadorDirector.java
import java.util.ArrayList;

public class CreadorDirector {
    private BuilderChimpoko builderChimpoko;
    CreadorDirector(BuilderChimpoko builder){
        builderChimpoko = builder;
    }
    Chinpokomon crearGallotronix(){
        ArrayList<Ataque> listaAtaques = new ArrayList<>();
        listaAtaques.add(crearGarraMecanica());

        builderChimpoko.setNombre("Gallotronix");
        builderChimpoko.setVida(25);
        builderChimpoko.setAtaques(listaAtaques);
        builderChimpoko.setNaturaleza(new Robot());
        return builderChimpoko.resultado();
    }
    Chinpokomon crearZapato(){
        ArrayList<Ataque> listaAtaques = new ArrayList<>();
        listaAtaques.add(crearZapatazo());

        builderChimpoko.setNombre("Zapato");
        builderChimpoko.setVida(30);
        builderChimpoko.setAtaques(listaAtaques);
        builderChimpoko.setNaturaleza(new Cosa());
        return builderChimpoko.resultado();
    }
    Chinpokomon crearZapatoEspecial(){
        ArrayList<Ataque> listaAtaques = new ArrayList<>();
        listaAtaques.add(crearZapatazo());
        listaAtaques.add(crearPomadaWassington());

        builderChimpoko.setNombre("ZapatoEspecial");
        builderChimpoko.setVida(30);
        builderChimpoko.setAtaques(listaAtaques);
        builderChimpoko.setNaturaleza(new Cosa());
        return builderChimpoko.resultado();
    }
    Chinpokomon crearCarnotron (){
        ArrayList<Ataque> listaAtaques = new ArrayList<>();
        listaAtaques.add(crearRayoVeloz());
        listaAtaques.add(crearCanionConico());

        builderChimpoko.setNombre("Carnotron");
        builderChimpoko.setVida(20);
        builderChimpoko.setAtaques(listaAtaques);
        builderChimpoko.setNaturaleza(new Animal());
        return builderChimpoko.resultado();
    }

    private Ataque crearGarraMecanica(){
        AtaqueConcret ataqueNuevo = new AtaqueConcret("Garra Mecanica", 2, 0, 2);
        return new ProbYDanioElevado( ataqueNuevo ,50.0,2);
    }
    private Ataque crearZapatazo(){
        AtaqueConcret ataqueNuevo = new AtaqueConcret("Zapatazo", 1, 0, 3);
        return new ProbYDanioVerdadero( ataqueNuevo ,10.0,50.0);
    }
    private Ataque crearPomadaWassington(){
        AtaqueConcret ataqueNuevo = new AtaqueConcret("Pomada Wassington", 0, 5,1);
        return new PomadaWassington(ataqueNuevo);
    }
    private Ataque crearRayoVeloz(){
        return new AtaqueConcret("Rayo Veloz", 3, 0,1);
    }
    private Ataque crearCanionConico(){
        return new AtaqueConcret("Canion Conico", 4, 0,1);
    }
}

File: python/CreadorDirector.py
from AtaqueConcrete import AtaqueConcret, Ataque
from Naturaleza import Robot, Cosa, Animal
from PomadaWassington import PomadaWassington
from ProbYDanioElevado import ProbYDanioElevado
from ProbYDanioVerdadero import ProbYDanioVerdadero


class CreadorDirector:
    def __init__(self, builder):
        self.__builder_chimpoko = builder
    
    def crear_gallotronix(self):
        self.__builder_chimpoko.set_nombre("Gallotronix")
        self.__builder_chimpoko.set_vida(25)
        self.__builder_chimpoko.set_ataques([self.crear_garra_mecanica()])
        self.__builder_chimpoko.set_naturaleza(Robot())
        return self.__builder_chimpoko.resultado()
    
    def crear_zapato(self):
        self.__builder_chimpoko.set_nombre("Zapato")
        self.__builder_chimpoko.set_vida(30)
        self.__builder_chimpoko.set_ataques([self.crear_zapatazo()])
        self.__builder_chimpoko.set_naturaleza(Cosa())
        return self.__builder_chimpoko.resultado()
    
    def crear_zapato_especial(self):
        self.__builder_chimpoko.set_nombre("ZapatoEspecial")
        self.__builder_chimpoko.set_vida(30)
        self.__builder_chimpoko.set_ataques([self.crear_zapatazo(), self.crear_pomada_wassington()])
        self.__builder_chimpoko.set_naturaleza(Cosa())
        return self.__builder_chimpoko.resultado()
    
    def crear_carnotron(self):
        self.__builder_chimpoko.set_nombre("Carnotron")
        self.__builder_chimpoko.set_vida(20)
        self.__builder_chimpoko.set_ataques([self.crear_rayo_veloz(), self.crear_canion_conico()])
        self.__builder_chimpoko.set_naturaleza(Animal())
        return self.__builder_chimpoko.resultado()
    
    def crear_rooster(self):
        self.__builder_chimpoko.set_nombre("Rooster")
        self.__builder_chimpoko.set_vida(60)
        self.__builder_chimpoko.set_ataques([self.crear_zapatazo(),self.crear_rayo_veloz(), self.crear_canion_conico(),self.crear_garra_mecanica(),self.crear_pomada_wassington()])
        self.__builder_chimpoko.set_naturaleza(Robot())
        return self.__builder_chimpoko.resultado()

    def crear_garra_mecanica(self):
        ataqueNuevo = AtaqueConcret("Garra Mecanica", 2, 0, 2)
        return ProbYDanioElevado( ataqueNuevo , 50, 2)
    
    def crear_zapatazo(self):
        ataqueNuevo = AtaqueConcret("Zapatazo", 1, 0, 3)
        return ProbYDanioVerdadero( ataqueNuevo ,10, 50)
    
    def crear_pomada_wassington(self):
        ataqueNuevo = AtaqueConcret("Pomada Wassington", 0, 5, 1)
        return PomadaWassington(ataqueNuevo)
    
    def crear_rayo_veloz(self):
        return AtaqueConcret("Rayo Veloz", 3, 0, 1)
    
    def crear_canion_conico(self):
        return AtaqueConcret("Canion Conico", 4, 0, 1)